Infrastructure/Services/ProcessEmailsService.cs

```diff
@@ -0,0 +1,450 @@
+            }
+
+            _logger.LogInformation(processedEmailsModel.Outcome);
+        }
+        catch (Exception ex)
+        {
+            await FinishProcessing(DateTime.UtcNow, $"Unexpected exception thrown. {ex}");
+            throw;
+        };
+
+        return processedEmailsModel;
+    }
+
+    public async Task<int> SendEmailAsync(int emailLogId)
+    {
+        return await SendEmailsToBeProcessedAsync(new int[] { emailLogId });
+    }
+
+    public async Task<int> SendEmailsAsync(int[] emailLogIds)
+    {
+        return await SendEmailsToBeProcessedAsync(emailLogIds);
+    }
+
+    public string? GetEmailAddressUsedForTesting(string? emailToBeUsedIfTestingEnabled = null)
+    {
+        string? testingEmail = null;
+
+        var overrideEmail = _emailSettingsConfig.OverrideAddress;
+
+        if (!string.IsNullOrWhiteSpace(overrideEmail))
+        {
+            testingEmail = overrideEmail;
+        }
+        else if (!string.IsNullOrWhiteSpace(emailToBeUsedIfTestingEnabled) && _emailSettingsConfig.AllSentToEnquirer)
+        {
+            testingEmail = emailToBeUsedIfTestingEnabled;
+        }
+
+        return testingEmail;
+    }
+
+    public EmailLog? MergeEmailForTesting(List<EmailLog> emailLogs, List<string> personalisationPropertiesToMerge)
+    {
+        if (_emailSettingsConfig.MergeResponses &&
+            emailLogs.Count > 1 &&
+            emailLogs.First().EmailPersonalisationLogs != null &&
+            personalisationPropertiesToMerge.Count > 0)
+        {
+            var initialEmailLog = emailLogs.First();
+            var initialEmailPersonalisationLogs = initialEmailLog!.EmailPersonalisationLogs!;
+
+            foreach (var personalisationPropertyToMerge in personalisationPropertiesToMerge)
+            {
+                var value = string.Empty;
+
+                var addNewLine = string.Empty;
+
+                foreach (var emailLog in emailLogs)
+                {
+                    var matchedEmailPersonalisationLog = emailLog!.EmailPersonalisationLogs!.Single(x => x.Key.Equals(personalisationPropertyToMerge));
+                    value = $"{value}{addNewLine}{emailLog.EmailAddress} - {matchedEmailPersonalisationLog.Value}";
+                    addNewLine = $"{Environment.NewLine}{Environment.NewLine}";
+                }
+
+                var initialEmailPersonalisationLog = initialEmailPersonalisationLogs.Single(x => x.Key.Equals(personalisationPropertyToMerge));
+                initialEmailPersonalisationLog.Value = value;
+            }
+
+            initialEmailPersonalisationLogs.Add(new EmailPersonalisationLog()
+            {
+                Key = TestExtraInfoKey,
+                Value = $"This is a merged email for testing purposes.{Environment.NewLine}{Environment.NewLine}"
+            });
+
+            initialEmailLog.EmailAddress = "merged_email_for_testing@education.gov.uk";
```

Comment:
We can refactor the line `initialEmailLog.EmailAddress = "merged_email_for_testing@education.gov.uk"; `by creating a constant or variable for the merged email address. This would make the code more flexible and easier to modify if the email address needs to be changed.